Assignment.Tests/PingProcessTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿using IntelliTect.TestTools;
+using System;
+using System.Diagnostics;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using static System.Runtime.InteropServices.JavaScript.JSType;
+
+namespace Assignment.Tests;
+#pragma warning disable CA1707 // Identifiers should not contain underscores
+[TestClass]
+public class PingProcessTests
+{
+    private PingProcess Sut { get; set; } = new();
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        Sut = new();
+    }
+
+    [TestMethod]
+    public void Start_PingProcess_Success()
+    {
+        Process process = Process.Start("ping", "localhost -c 4");
+        process.WaitForExit();
+        Assert.AreEqual<int>(0, process.ExitCode);
+    }
+
+
+    //External pings seem to not work in github
+    //[TestMethod]
+
+    //public void Run_GoogleDotCom_Success()
+    //{
+    //    int exitCode = Sut.Run("google.com -c 4").ExitCode;
+    //    Assert.AreEqual<int>(0, exitCode);
+    //}
+
+
+    [TestMethod]
+    public void Run_InvalidAddressOutput_Success()
+    {
+        (int exitCode, string? stdOutput) = Sut.Run("badaddress");
+        stdOutput = WildcardPattern.NormalizeLineEndings(stdOutput!.Trim());
+        Assert.IsTrue(string.IsNullOrWhiteSpace(stdOutput));
+
+        //On linux this does not happen
+        //Assert.AreEqual<string?>(
+        //    "Ping request could not find host badaddress. Please check the name and try again.".Trim(),
+        //    stdOutput,
+        //    $"Output is unexpected: {stdOutput}");
+        Assert.AreEqual<int>(2, exitCode);
+    }
+
+    [TestMethod]
+    public void Run_CaptureStdOutput_Success()
+    {
+        PingResult pingResult = Sut.Run("localhost -c 4");
+        Assert.AreEqual(0, pingResult.ExitCode);
+    }
+
+    [TestMethod]
+    [DataRow("localhost -c 4")]
+    public void RunTaskAsync_Success(string address)
+    {
+
+        //Act
+        Task<PingResult> task = Sut.RunTaskAsync(address);
+        task.Wait();
+        //Assert
+        Assert.AreEqual(0, task.Result.ExitCode);
+    }
+
+    [TestMethod]
+
+    [DataRow("localhost -c 4")]
+    public void RunAsync_UsingTaskReturn_Success(string address)
+    {
+        //Arrange
+
+        //Act
+
+        Task<PingResult> pingResult = Sut.RunAsync(address);
+        pingResult.Wait();
+        //Assert
+
+        Assert.AreEqual(0, pingResult.Result.ExitCode);
+
+    }
+
+    [TestMethod]
+    [DataRow("localhost -c 4")]
+    public async Task RunAsync_UsingTpl_Success(string hostName)
+    {
+        // Arrange & Act
+        Task<PingResult> taskResult = Sut.RunAsync(hostName);
+        PingResult pingResult = await taskResult;
```

Comment:
Nit: We can simplify this. Generally in async await, we await the calls inline
```suggestion
        PingResult pingResult = await Sut.RunAsync(hostName);
```